src/gamepad.cpp

```diff
@@ -0,0 +1,217 @@
+                printf("buttons--------------------------------------------\n");
+                f>>t;
+                buttons[t] = BUTTON_LB;
+                f>>t;
+                buttons[t] = BUTTON_RB;
+                f>>t;
+                buttons[t] = BUTTON_SELECT;
+                f>>t;
+                buttons[t] = BUTTON_START;
+                f>>t;
+                buttons[t] = BUTTON_LS;
+                f>>t;
+                buttons[t] = BUTTON_RS;
+            }
+            if(s == "[dpad]") {
+                printf("dpad-------------------------------------------------\n");
+                f>>t;
+                DPAD_LR = t;
+                f>>t;
+                DPAD_LR = t;
+            }
+            if(s == "[axis]") {
+                printf("axis---------------------------------------------------\n");
+                f>>t;
+                LAXIS_X = t;
+                f>>t;
+                LAXIS_Y = t;
+                f>>t;
+                RAXIS_X = t;
+                f>>t;
+                RAXIS_Y = t;
+                f>>t;
+                MUTUAL_AXIS = t;   
+            }
+            s="";
+        }
+        f.close();
+        GameControllerManager::sGamePadManager->setGameControllerConnected(0, true);
+    } 
+    else 
+        printf("open failed.\n");
+}
+
+
+void LinuxGamepadManager::pool() {
+    //main loop important to do everything there
+    float val;
+    JoystickEvent event;   
+
+    while (true)
+    {
+        // Attempt to sample an event from the joystick
+        
+        if (joystick->sample(&event))
+        {
+            if (event.isButton())
+            {
+                //printf("Button %u is %s\n",event.number,event.value == 0 ? "up" : "down");
+                //usage buttons
+                if(event.number == 6 || event.number == 7){
+                    GameControllerManager::sGamePadManager->feedTrigger(0, buttons[event.number], (float) event.value);
+                } else 
+                    GameControllerManager::sGamePadManager->feedButton(0, buttons[event.number], event.value, true);
+                break;
+ 
+            }
+            else if (event.isAxis())
+            {
+                //printf("Axis %u is at position %d\n", event.number, event.value);
+                if(event.number == DPAD_LR) { //dpad usage
+                    if(event.value > 0) lastdpad[0] = BUTTON_DPAD_RIGHT;
+                    if(event.value < 0) lastdpad[0] = BUTTON_DPAD_LEFT;
+                    GameControllerManager::sGamePadManager->feedButton(0, lastdpad[0], event.value !=0 ? 1 : 0, true);
+                }
+                if(event.number == DPAD_UD) { //dpad usage
+                    if(event.value > 0) lastdpad[1] = BUTTON_DPAD_DOWN;
+                    if(event.value < 0) lastdpad[1] = BUTTON_DPAD_UP;
+                    GameControllerManager::sGamePadManager->feedButton(0, lastdpad[1], event.value !=0 ? 1 : 0, true);
+                }
+
```

Comment:
There is anyway to configure the axis? I don't think Minecraft handle that right?

Reply:
You can reconfigure it there but not in game